src/WebJobs.Script/Config/RouteHandlingOptions.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using AuthLevel = Microsoft.Azure.WebJobs.Extensions.Http.AuthorizationLevel;
+
+namespace Microsoft.Azure.WebJobs.Script.Configuration
+{
+    /// <summary>
+    /// Defines the routing mode for HTTP requests.
+    /// </summary>
+    public enum RouteHandlingMode
+    {
+        /// <summary>
+        /// Routes are mapped based on httpTrigger functions defined in function metadata (default).
+        /// </summary>
+        Function,
+
+        /// <summary>
+        /// Creates a single catch-all route that handles all requests and proxies them to the custom handler.
+        /// </summary>
+        All
+    }
+
+    public class RouteHandlingOptions
+    {
+        private AuthorizationLevel? _authorizationLevel;
+
+        /// <summary>
+        /// Mode determining how routes are mapped.
+        /// </summary>
+        public RouteHandlingMode Mode { get; set; } = RouteHandlingMode.Function;
+
+        /// <summary>
+        /// Only applicable to mode = "all". Determines the authorization level for the catch-all route.
+        /// Defaults to "function" when Mode == "all". When Mode == "function" this defaults to null.
+        /// Supports all AuthorizationLevel values: Anonymous, Function, User, Admin, System.
+        /// </summary>
+        public AuthorizationLevel? AuthorizationLevel
+        {
+            get => _authorizationLevel ??
+                   (Mode == RouteHandlingMode.All ? AuthLevel.Function : null);
+            set => _authorizationLevel = value;
+        }
+
+        /// <summary>
+        /// Gets the AuthorizationLevel for binding purposes.
+        /// </summary>
+        /// <returns>The AuthorizationLevel to use for binding.</returns>
+        public AuthorizationLevel GetAuthorizationLevelForBinding()
+        {
+            return AuthorizationLevel ?? AuthLevel.Function;
+        }
+
+        /// <summary>
+        /// Gets the authorization level as a string for compatibility.
+        /// </summary>
+        /// <returns>The authorization level as a string, or null if not set.</returns>
+        public string GetAuthorizationLevelString()
+        {
+            var authLevel = AuthorizationLevel;
+            return authLevel?.ToString().ToLowerInvariant();
+        }
+
+        /// <summary>
+        /// Creates HTTP handler metadata for the catch-all route when mode is "all".
+        /// </summary>
+        /// <returns>A FunctionMetadata object representing the HTTP handler.</returns>
+        public Description.FunctionMetadata CreateHttpHandlerMetadata()
```

Comment:
Can create an implementation of [IFunctionsProvider](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.Abstractions/Description/IFunctionProvider.cs) to generate and inject this metadata.